src/app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-server.cpp

```diff
@@ -163,7 +164,7 @@ CHIP_ERROR WiFiDiagosticsAttrAccess::ReadWiFiRssi(AttributeValueEncoder & aEncod
     {
         rssi.SetNonNull(value);
         ChipLogProgress(Zcl, "The current RSSI of the Node’s Wi-Fi radio in dB: %d", value);
-        MATTER_TRACE_METRIC("wifi_rssi", value);
+        MATTER_LOG_METRIC(WiFiRSSI, value);
```

Comment:
This seems to hide away that metric names need to be in `src/tracing/metric_keys.h`.

Is there a reason we have metric_keys.h globally instead of the previous support of free-form strings? I believe less global see-everywhere headers/constants is better.

Reply:
Please use strings or constants rather than stringifying IDs in-place. This avoid metrics IDs looking like variable identifdeirs or shadowing them. For example, here, just use `kMetricWiFiRSSI` rather than hiding the `kMetric` prefix.

Reply:
Free form strings don't work for metrics, they can't ever change, otherwise metrics break.

Reply:
@anush-apple Can we prefix these with k? kWiFiRSSI

Reply:
@anush-apple I think, even if we decide to use a fixed set of strings for the common SDK part, we should still support free-form strings in the API. This can be very useful for any application or platform specific metrics.

Reply:
My new changes will not prevent free forms strings. But we should consider documenting any metric that is part of the common core code to ensure these keys are not modified once deployed.

Reply:
We can have lints about modification of particular files to require additional audit.

Reply:
> I also prefer using the full string content. I followed the pattern used in ChipLogProgress which was not using the full string for the module name. Perhaps we should visit that as well in some distant future. I will fix the code to switch to full string constants names, which is my preference as well.

Yes, the pattern of `ChipLogXxxx` is a bit unfortunate, and I really think it should not be repeated here.

BTW having all the constants be used "in-place" in the call sites allows for "find everywhere a metric is used in code` via `kMetric[A-Za-z0-9]+` regex, which has enough specificity to help track through the whole code. 

Having those names also ensures that CTRL-click to follow a symbol in every IDE works as intended. When something goes through token concat within a macro to generate a symbol, most IDEs cannot cope to follow.

Reply:
I can say that I lost multiple hours (days)? to re-figure out each time what possible values ChipLog can take beside the ones I use the most like App or Discovery. it is a horrible pattern from my perspective. 

To quickly try: I give you `ChipLogProgress(App, "test")`. How fast can a new developer tell me what other things besides `App` is supported there? Even today it takes me a few minutes to figure out.

Reply:
I would also say: the nature of tracing is that things get added, however very rarely renamed (maybe because typos? however those should have been caught during review). This is why it still works even when allowing free-form strings.

Reply:
Switched to supporting free forms strings. Made the macro require complete string constant name and no more PP prefixing